### PR TITLE
Enhance input validation, CLI functionality, and testing for directLFQ

### PR DESCRIPTION
**Pull request by ammarcsj**
This PR adds duplicate row handling, fixes a non-functional CLI option, expands quicktests with a CLI run, updates GitHub Actions workflows, and bumps the version to 0.2.20.

- Duplicates in input
    - Added a check to raise an error when duplicate rows exist in the input dataframe
    - Implemented removal of potential duplicate quant_id values, with logging of removed entries 
- CLI improvements
    - Fixed the --filter_dict option in the CLI, setting default to None instead of False, changing the type from bool to str and correcting the short flag from -dn to -fd
- Workflow updates
    - Added the miniconda-version: latest parameter to all GitHub Actions workflows using conda
- Version bump
    - Incremented version from 0.2.19 to 0.2.20 across all relevant files
- Added CLI run to quicktests
    - Included a directLFQ CLI run in the quicktests script for more comprehensive testing

### CODE REVIEW

File: directlfq/utils.py [hunk 1]
```diff
@@ -310,6 +310,17 @@ def index_and_log_transform_input_df(data_df):
 def remove_allnan_rows_input_df(data_df):
     return data_df.dropna(axis = 0, how = 'all')
 
+def remove_potential_quant_id_duplicates(data_df):
```
Comment (mschwoer) [inline directlfq/utils.py:313]: type hints & docstring?

File: directlfq/utils.py [hunk 2]
```diff
@@ -310,6 +310,17 @@ def index_and_log_transform_input_df(data_df):
...
+    before_drop = len(data_df)
+    data_df = data_df.drop_duplicates(subset=config.QUANT_ID, keep='first')
+    after_drop = len(data_df)
+    if before_drop != after_drop:
+        entries_removed = before_drop - after_drop
+        LOGGER.info(f"Duplicate quant_ids detected. {entries_removed} rows removed from input df.")
```
Comment (mschwoer) [inline directlfq/utils.py:319]: warning?

File: directlfq/utils.py [hunk 3]
```diff
@@ -310,6 +310,17 @@ def index_and_log_transform_input_df(data_df):
 def remove_allnan_rows_input_df(data_df):
     return data_df.dropna(axis = 0, how = 'all')
 
+def remove_potential_quant_id_duplicates(data_df):
+    before_drop = len(data_df)
+    data_df = data_df.drop_duplicates(subset=config.QUANT_ID, keep='first')
+    after_drop = len(data_df)
+    if before_drop != after_drop:
+        entries_removed = before_drop - after_drop
```
Comment (mschwoer) [inline directlfq/utils.py:318]: (nit) could be simplified
`if (entries_removed:= before_drop - after_drop):`

**Review by mschwoer — APPROVED** [review]
LGTM

File: directlfq/utils.py (hunk 1 above)
Comment (ammarcsj) [reply directlfq/utils.py:313]: 👍

File: directlfq/utils.py (hunk 2 above)
Comment (ammarcsj) [reply directlfq/utils.py:319]: 👍

**Comment by ammarcsj** [comment]
thanks Magnus!

**Pull request merged**